src/librustc/diagnostics.rs

```diff
@@ -1944,6 +1944,38 @@ you want. Example:
 ```
 "##,
 
+E0492: r##"
+A borrow of a constant containing interior mutability was attempted. Erroneous
+code example:
+
+```
+#![feature(const_fn)]
+
+use std::cell::Cell;
+
+const A: Cell<usize> = Cell::new(1);
+const B: &'static Cell<usize> = &A;
+// error: cannot borrow a constant which contains interior mutability, create
+//        a static instead
+
+// or:
+struct C { a: Cell<usize> }
+
+const D: C = C { a: Cell::new(1) };
+const E: &'static Cell<usize> = &D.a; // error
+
+// or:
+const F: &'static C = &D; // error
+
+fn main() {}
+```
+
+To solve this error, either use statics or avoid using `cell` type. If you want
```

Comment:
So... I think we should have an example of things that do work, if we're going to suggest something as a solution. There can be text and/or more examples discussing why this solution doesn't work all the time, but that shouldn't be a reason to not suggest things that work.

IMO, the goals for these sort of explanations are:

- making it clear exactly what the problem is
- explaining why something is an error (when it makes sense, like here)
- helping people with a solution, as much as possible (code examples are important)